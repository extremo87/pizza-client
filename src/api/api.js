import axios from 'axios';


const ServerResponseCode = {
  UNAUTHORIZED: 401,
  VALIDATION: 422
};

export const createAPI = (onError) => {
  const api = axios.create({
    baseURL: process.env.API_URL,
    timeout: 5000
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (error) => {
    if (error.response) {
      if (error.response.status === ServerResponseCode.VALIDATION) {
        const errors = [];

        Object.keys(error.response.data.errors).map((key) => {
          errors.push(error.response.data.errors[key]);
        });

        onError(errors.join(`,`));

      } else {
        onError(error.message);
      }
    } else {
      onError(error.message);
    }
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
