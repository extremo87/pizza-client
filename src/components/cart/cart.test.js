import React from 'react';
import Cart from './cart';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import NameSpace from "../../reducers/name-space.js";
import thunk from 'redux-thunk';
import {createAPI} from "../../api/api";
import {BrowserRouter} from 'react-router-dom';


const api = createAPI(() => {});

const mockStore = configureStore([thunk.withExtraArgument(api)]);


const cart = {
  1: {
    id: 1,
    title: `MARGHERITA`,
    ingredients: `Tomatoes, mozzarella, ricotta, bacon, onion`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    image: `https://static.nnc.kz/pizza1.jpg`,
    price: 18,
    created_at: `2020-09-21 13:18:35`,
    updated_at: `2020-09-21 13:18:35`,
    qty: 2
  },
  2: {
    id: 2,
    title: `QUATTRO FROMAGGI`,
    ingredients: `Gorgonzola, ricotta, mozzarella, taleggio`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    image: `https://static.nnc.kz/pizza2.jpg`,
    price: 24,
    created_at: `2020-09-21 13:18:35`,
    updated_at: `2020-09-21 13:18:35`,
    qty: 1
  }
};

const store = mockStore({
  [NameSpace.CART]: {
    cart,
  },
  [NameSpace.DATA]: {
    currency: `EUR`,
    deliveryFee: 4,
    currencies: {
      USD: {
        rates: {
          EUR: {
            symbol: `€`,
            rate: 1.18
          }
        },
        symbol: `$`
      },
      EUR: {
        rates: {
          USD: {
            symbol: `$`,
            rate: 0.84
          }
        },
        symbol: `€`
      }
    }
  }
});


it(`Cart conponent renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Cart
              cart={cart}
              deliveryFee={5}
              key={1}
              currency={`USD`}
            />
          </BrowserRouter>
        </Provider>

    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
