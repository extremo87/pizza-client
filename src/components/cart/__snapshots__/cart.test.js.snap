// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart conponent renders correctly 1`] = `
<div>
  <h4
    className="d-flex justify-content-between align-items-center mb-3"
  >
    <span
      className="text-muted"
    >
      Your cart
    </span>
    <span
      className="badge badge-secondary badge-pill"
    >
      3
    </span>
  </h4>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item d-flex justify-content-between lh-condensed"
    >
      <div>
        <h6
          className="my-0"
        >
          MARGHERITA
        </h6>
        <small
          className="text-muted"
        >
          Tomatoes, mozzarella, ricotta, bacon, onion
        </small>
      </div>
      <div
        style={
          Object {
            "left": "70%",
            "position": "absolute",
          }
        }
      >
        <div
          className="col-mb-2"
        >
          <div
            className="input-group"
          >
            <div
              className="input-group-prepend"
            >
              <button
                className="btn btn-sm btn-outline-secondary"
                onClick={[Function]}
                type="button"
              >
                +
              </button>
            </div>
            <div
              className="custom-file"
              style={
                Object {
                  "width": "40px",
                }
              }
            >
              <input
                className="form-control"
                readOnly={true}
                type="text"
                value={2}
              />
            </div>
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-sm btn-outline-secondary"
                onClick={[Function]}
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
      <span
        className="text-muted"
      >
        36
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between lh-condensed"
    >
      <div>
        <h6
          className="my-0"
        >
          QUATTRO FROMAGGI
        </h6>
        <small
          className="text-muted"
        >
          Gorgonzola, ricotta, mozzarella, taleggio
        </small>
      </div>
      <div
        style={
          Object {
            "left": "70%",
            "position": "absolute",
          }
        }
      >
        <div
          className="col-mb-2"
        >
          <div
            className="input-group"
          >
            <div
              className="input-group-prepend"
            >
              <button
                className="btn btn-sm btn-outline-secondary"
                onClick={[Function]}
                type="button"
              >
                +
              </button>
            </div>
            <div
              className="custom-file"
              style={
                Object {
                  "width": "40px",
                }
              }
            >
              <input
                className="form-control"
                readOnly={true}
                type="text"
                value={1}
              />
            </div>
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-sm btn-outline-secondary"
                onClick={[Function]}
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
      <span
        className="text-muted"
      >
        24
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between bg-light"
    >
      <div
        className="text-success"
      >
        <h6
          className="my-0"
        >
          Delivery fee
        </h6>
      </div>
      <span
        className="text-success"
      >
        €
        4.00
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between"
    >
      <span>
        Total (
        EUR
        )
      </span>
      <strong>
        €
        64.00
         
      </strong>
    </li>
  </ul>
  <hr
    className="mb-4"
  />
  <a
    className="btn btn-primary btn-lg btn-block"
    href="/checkout"
    onClick={[Function]}
    type="submit"
  >
    Continue to checkout
  </a>
</div>
`;
